Remove special `quirk` mode handling of `labels` with `option` element

Ahmad-S792 opened the pull request:
#### 7f73e6048a4592575300e0781d1b69277801876f
<pre>
Remove special `quirk` mode handling of `labels` with `option` element
<a href="https://bugs.webkit.org/show_bug.cgi?id=293543">https://bugs.webkit.org/show_bug.cgi?id=293543</a>

Reviewed by Aditya Keerthi.

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium.

This patch removes special case of `quirk` mode of `labels` with `option`
element and now will use `label` attribute matching &apos;standards&apos; mode, which
was previously ignored.

It was discussed in following github issue [1]:

[1] <a href="https://github.com/whatwg/html/issues/2988">https://github.com/whatwg/html/issues/2988</a>

Blink rolled out this change in 2023 and haven&apos;t reverted behavior to
reflect that there were not any compatibility issues, if there would be
in case of WebKit, it is easier to rollout as well.

Unfortunately, we cannot enable corresponding WPT test case yet [2] since
we still have 0.07% pixel differences due to borders of &apos;select&apos; element
but we do match now other browser on showing labels inside.

[2] html/rendering/widgets/the-select-element/option-add-label-quirks.html

* Source/WebCore/html/HTMLOptionElement.cpp:
(WebCore::HTMLOptionElement::textIndentedToRespectGroupLabel const):
(WebCore::HTMLOptionElement::displayLabel const): Deleted.
* Source/WebCore/html/HTMLOptionElement.h:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::focusedElementInformation):

It is good to keep these local tests till we don&apos;t progress WPT properly,
this is just updating test expectations to match new behavior.
* LayoutTests/fast/dom/HTMLOptionElement/option-label-quirksmode-expected.html:
* LayoutTests/fast/dom/HTMLOptionElement/option-label-quirksmode.html:
* LayoutTests/fast/dom/HTMLOptionElement/option-label-quirksmode2-expected.html:
* LayoutTests/fast/dom/HTMLOptionElement/option-label-quirksmode2.html:
* LayoutTests/fast/forms/ios/select-option-label-quirks-mode-expected.txt:
* LayoutTests/fast/forms/ios/select-option-label-quirks-mode.html:

Canonical link: <a href="https://commits.webkit.org/295503@main">https://commits.webkit.org/295503@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1cec4a1aee19e5ed5388fac5a57e46e3a8b28dc4

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/105244 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/24956 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/15381 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/110457 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/55902 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/107285 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/25386 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/33500 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/79938 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/108250 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/19792 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/94990 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/60240 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/19546 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/13064 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/55298 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/89247 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/13108 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/113052 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/32404 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/23869 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/89012 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/32768 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/91206 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/88649 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22612 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/33541 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/11327 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/27813 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/32327 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/37739 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/32106 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/35451 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/33674 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on LayoutTests/fast/dom/HTMLOptionElement/option-label-quirksmode-expected.html:
```diff
@@ -2,7 +2,7 @@
 <body>
 <p>Tests that the option's label is not displayed in quirks mode.</p>
```
This description is no longer accurate right?

Ahmad-S792 replied:
@pxlcoder - Good catch! Updated test + expected files + rebased as well.

Ahmad-S792 committed:
Remove special `quirk` mode handling of `labels` with `option` element
https://bugs.webkit.org/show_bug.cgi?id=293543

Reviewed by Aditya Keerthi.

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium.

This patch removes special case of `quirk` mode of `labels` with `option`
element and now will use `label` attribute matching 'standards' mode, which
was previously ignored.

It was discussed in following github issue [1]:

[1] https://github.com/whatwg/html/issues/2988

Blink rolled out this change in 2023 and haven't reverted behavior to
reflect that there were not any compatibility issues, if there would be
in case of WebKit, it is easier to rollout as well.

Unfortunately, we cannot enable corresponding WPT test case yet [2] since
we still have 0.07% pixel differences due to borders of 'select' element
but we do match now other browser on showing labels inside.

[2] html/rendering/widgets/the-select-element/option-add-label-quirks.html

* Source/WebCore/html/HTMLOptionElement.cpp:
(WebCore::HTMLOptionElement::textIndentedToRespectGroupLabel const):
(WebCore::HTMLOptionElement::displayLabel const): Deleted.
* Source/WebCore/html/HTMLOptionElement.h:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::focusedElementInformation):

It is good to keep these local tests till we don't progress WPT properly,
this is just updating test expectations to match new behavior.
* LayoutTests/fast/dom/HTMLOptionElement/option-label-quirksmode-expected.html:
* LayoutTests/fast/dom/HTMLOptionElement/option-label-quirksmode.html:
* LayoutTests/fast/dom/HTMLOptionElement/option-label-quirksmode2-expected.html:
* LayoutTests/fast/dom/HTMLOptionElement/option-label-quirksmode2.html:
* LayoutTests/fast/forms/ios/select-option-label-quirks-mode-expected.txt:
* LayoutTests/fast/forms/ios/select-option-label-quirks-mode.html:

Canonical link: https://commits.webkit.org/295503@main
